src/ChainSafe.Gaming.MetaMask.Unity/MetaMaskController.cs

```diff
@@ -56,7 +61,7 @@ public void Initialize(ILogWriter logWriter)
         /// <returns>Connected account address.</returns>
         public Task<string> Connect()
         {
-            var taskCompletionSource = new TaskCompletionSource<string>();
+            ConnectedTsc = new TaskCompletionSource<string>();
```

Comment:
Before setting this `tcs` you should first check if it's already set and it isn't complete to set and exception with `"Already trying to connect"` and return that task